Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -200,24 +189,40 @@ CGPathRef CGPathCreateCopy(CGPathRef path) {
 }
 
 /**
- @Status Interoperable
+@Status Interoperable
+@Notes
 */
 CGMutablePathRef CGPathCreateMutableCopy(CGPathRef path) {
     if (path == NULL) {
         return NULL;
     }
 
-    auto ret = __CGPath::alloc(nil);
-    ret->_max = path->_max;
-    ret->_count = path->_count;
-    ret->_elements = (CGPathElementInternal*)IwRealloc(ret->_elements, ret->_max * sizeof(CGPathElementInternal));
-    memcpy(ret->_elements, path->_elements, path->_count * sizeof(CGPathElementInternal));
-    // All of the new elements need to call init
-    // because they were created with alloc + memcpy
-    for (unsigned i = 0; i < path->_count; i++) {
-        ret->_elements[i].init();
+    size_t memSize = sizeof(struct __CGPath) - sizeof(CFRuntimeBase);
+    CGPathRef ret = ((CGPathRef)(_CFRuntimeCreateInstance(kCFAllocatorDefault, CGPathGetTypeID(), memSize, NULL)));
+    struct __CGPath* mutableRet = const_cast<struct __CGPath*>(ret);
+
+    ComPtr<ID2D1Factory> factory = _CreateD2DFactoryInstance();
+
+    if (FAILED(factory->CreatePathGeometry(&mutableRet->_pathGeometry))) {
+        CFRelease(mutableRet);
+        return nullptr;
+    }
+    if (FAILED(mutableRet->_pathGeometry->Open(&mutableRet->_geometrySink))) {
+        CFRelease(mutableRet);
+        return nullptr;
     }
-    return ret;
+
+    __closePath(path);
+    if (FAILED(path->_pathGeometry->Stream(mutableRet->_geometrySink.Get()))) {
+        CFRelease(mutableRet);
+        return nullptr;
+    }
+
+    mutableRet->_currentPoint = { path->_currentPoint.x, path->_currentPoint.y };
+    // Always create an open copy as opening a closed copy would be more expensive.
```

Comment:
Don't forget to copy the source point!

Also, you should be able to do `mutableRet->point = path->point` instead of going member by member